### PR TITLE
BKNDLSS-30119 Tabs

### PR DESCRIPTION
**Pull request by Saharij**
Screenshot 2022-10-24 at 14 50 56
Screenshot 2022-10-24 at 14 52 22
Screenshot 2022-10-24 at 14 54 27

### CODE REVIEW

File: components/bl-tabs-component/src/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div className={ cn("bl-customComponent-tabs", classList, { disabled }) } style={ style }>
+      <div className="tabs">
+        { tabsList.map(({ id, label }) => (
+          <Tab
```
Comment (v-excelsior) [inline components/bl-tabs-component/src/index.js:66]: TabContorl

File: components/bl-tabs-component/src/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    if (!currentTab) {
+      setCurrentTabId(tabsList[0] && tabsList[0].id || null);
+    }
+  }, [currentTabId, tabsList]);
+
+  useEffect(() => {
+    podsContent.dataModel.currentTabId = currentTabId;
+  }, [currentTabId]);
+
+  const handleClickTab = id => {
```
Comment (v-excelsior) [inline components/bl-tabs-component/src/index.js:50]: `onChange` or `onTabChange`

File: components/bl-tabs-component/styles/index.less [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@bl-customComponent-tabs-padding: 12px 16px;
+@bl-customComponent-tabs-backgroundColor: rgba(@themePrimary, 0.15);
+
+.bl-customComponent-tabs {
+  .tabs {
+    display: flex;
+    justify-content: center;
+  }
+
+  .tab {
```
Comment (v-excelsior) [inline components/bl-tabs-component/styles/index.less:16]: Could we add some different style variants for tab controls?

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-tabs-component/src/index.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  const podsContent = pods["Tabs Content"];
+
+  const tabsList = useMemo(() => {
+    if (Array.isArray(tabs)) {
+      return tabs.filter((tab, index) => {
+        if (tab.id && tab.label) {
+          return true;
+        }
+
+        console.log(`Invalid tab item ${++index}`);
```
Comment (v-excelsior) [inline components/bl-tabs-component/src/index.js:28]: console.error

`Invalid tab item: {label:..., value:...}` or something readable, but not shifted index.

File: components/bl-tabs-component/src/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+        return false;
+      });
+    }
+
+    return [];
+  }, [tabs]);
```
Comment (v-excelsior) [inline components/bl-tabs-component/src/index.js:35]:
```suggestion
  const tabsList = useMemo(() => {
    if (!Array.isArray(tabs)) {
      return []
    }

    return ....
  }, [tabs]);
```

File: components/bl-tabs-component/src/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    setCurrentTabId(id);
+    onChange({ currentTabId: id });
+  }
+
+  const podsContent = pods["Tabs Content"];
+
+  const tabsList = useMemo(() => {
+    if (Array.isArray(tabs)) {
+      return tabs.filter((tab, index) => {
+        if (tab.id && tab.label) {
```
Comment (v-excelsior) [inline components/bl-tabs-component/src/index.js:24]: Empty string is valid label for me.

File: components/bl-tabs-component/src/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+  const [currentTabId, setCurrentTabId] = useState(null);
+
+  component.getCurrentTabId = () => currentTabId;
+  component.setCurrentTabId = id => {
+    setCurrentTabId(id);
+    onChange({ currentTabId: id });
+  }
+
+  const podsContent = pods["Tabs Content"];
```
Comment (v-excelsior) [inline components/bl-tabs-component/src/index.js:19]: IDE doesn't fix double quotes here?

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-tabs-component/src/index.js (hunk 7 above)
Comment (Saharij) [reply components/bl-tabs-component/src/index.js:19]: Fixes it, but I forgot to run editorconfig.

File: components/bl-tabs-component/src/index.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+    return tabs.filter(({ id, label, value }) => {
+      if (id) {
+        return true;
+      }
+
+      console.error(`Invalid tab item: {label: ${label}, value: ${value}}`);
+
+      return false;
+    });
```
Comment (v-excelsior) [inline components/bl-tabs-component/src/index.js:39]: I made mistake, there no any value as I see.

```
console.error(`Tab ID must be provided. Invalid tab item: {label: ${label}, id: ${id}}`);
```

**Pull request merged**